src/Forms/Prism.Forms/Services/DeviceService.cs

```diff
@@ -66,6 +67,30 @@ public void BeginInvokeOnMainThread(Action action)
             Device.BeginInvokeOnMainThread(action);
         }
 
+        /// <summary>
+        /// Invokes an action (which can be awaited) on the device main UI thread.
+        /// </summary>
+        /// <param name="action">The Action to invoke</param>
+        public Task BeginInvokeOnMainThreadAsync(Action action)
+        {
+            var tcs = new TaskCompletionSource<bool>();
+            
+            Device.BeginInvokeOnMainThread(() =>
+            {
+                try
+                {
+                    action();
+                    tcs.TrySetResult(true);
+                }
+                catch (Exception ex)
+                {
+                    tcs.TrySetException(ex);
+                }
+            });
+            
+            return tcs.Task;
+        }
```

Comment:
This is not something that we will be adding. Xamarin.Forms has an `InvokeOnMainThreadAsync` which we can map to, but we're not going to add an entirely new implementation.

Reply:
Ok. Did I miss something then? Should I base this on the shell branch, as I’m assuming that’s built against a newer version of XF?

Reply:
go ahead and update to 4.5. You'll see the new method signatures on the Device class there

Reply:
Great. I was just being a little wary, I didn’t know if there was a specific reason that it hadn’t been updated. I’ll get that sorted. Thanks for the clarification. 👍

Reply:
We don't update Xamarin.Forms just because there are updates available. That said Prism 8 will be updating as there are some changes that we will adjusting for

Reply:
Makes complete sense. I’ll bear that in mind going forward. I’m in the U.K, so it’s bedtime for me now. I’ll get this sorted tomorrow. Thanks for your time.

Reply:
OK. I've made the changes now.

I did notice a "Device.GetMainThreadSynchronizationContextAsync" method has been added to the XF device class too. Should support also be provided for this? If so, I'm happy to do that as well.

Reply:
anything that no longer maps properly should be removed, and any new additions can be added to the DeviceService